libraries/AP_RCProtocol/AP_RCProtocol_DSM.cpp

```diff
@@ -236,13 +241,13 @@ bool AP_RCProtocol_DSM::dsm_decode(uint32_t frame_time_ms, const uint8_t dsm_fra
 
     // Handle VTX control frame.
     if (haveVtxControl) {
-#if !APM_BUILD_TYPE(APM_BUILD_iofirmware)
+#if AP_RCPROTOCOL_SRXL2_ENABLED
```

Comment:
This is not correct. This works even if you are not using SRXL2 - probably this function needs to be split out somehow

Reply:
So I'm breaking stuff here.  That's not good.

Is there some conceptual reason I'm missing here that says `configure_vtx` shouldn't go into AP_VideoTX?

Failing that I could move it up into `AP_RCProtocol_Backend`....

Reply:
It's Spektrum specific, so only applies to SRXL2, DSM (and possibly SRXL)

Reply:
I've moved it into AP_RCProtocol_Backend with a new patch on top.  Maybe the method will get elided as required.  Even if it doesn't, it's better not perfect :-)

If we wanted to be fancy we could do some sort of mixin  thingy instead (`class AP_RCProtocol:SRXL2 : public AP_RCProtocol_backend, protected AP_RCProtocol_backend_has_spektrum_vtx_control`), but it's not a pattern we use much.